src/actors/common/cache.rs

```diff
@@ -273,19 +281,29 @@ impl<T: CacheItemRequest> Cacher<T> {
         }));
 
         // Run the computation and wrap the result in Arcs to make them clonable.
-        let channel = future::lazy(move || slf.compute(request, key))
-            .then(move |result| {
-                // Drop the token first to evict from the map. This ensures that callers either
-                // get a channel that will receive data, or they create a new channel.
-                drop(remove_computation_token);
-                sender.send(result.map(Arc::new).map_err(Arc::new)).ok();
-                Ok(())
-            })
-            .sentry_hub_new_from_current();
+        let channel = async move {
+            let result = match slf.compute(request, key).await {
+                Ok(ok) => Ok(Arc::new(ok)),
+                Err(err) => Err(Arc::new(err)),
+            };
+            drop(remove_computation_token);
+            sender.send(result).ok();
+            Ok(())
+        }
+        .bind_hub(Hub::new_from_top(Hub::current()));
+        // let channel = future::lazy(move |_| slf.compute(request, key))
+        //     .then(move |result| {
+        //         // Drop the token first to evict from the map. This ensures that callers either
```

Comment:
please move the comment to the right place

Reply:
I forgot to remove this whole block, it was the alternative that I didn't get to work.  The async move block above does the same.

Reply:
huh, never mind.  you're right and i re-added the comment.